src/components/LocationSearch/LocationSearch.test.jsx

```diff
@@ -0,0 +1,54 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import LocationSearch from "./LocationSearch";
+import { describe, it, expect, vi } from "vitest";
+
+describe("</ LocationSearch", () => {
+  const props = {
+    error: "",
+    locations: ["New York", "São Paulo"],
+    isLocationSearchOpen: true,
+    handleLocationSearch: vi.fn(),
+    handleLocationSearchVisibility: vi.fn(),
+  };
+
+  it("renders the error message", () => {
+    render(<LocationSearch {...props} error="some error here" />);
```

Comment:
Instead of creating this way, you could have a method like

```jsx
const renderLocationSearch = (props) => render(<LocationSearch {...props} />)
```

Having the default props you just created and then passing the ones you need